### ISSUE TITLE
[Android] `flutter run --multidex` fails with error: package androidx.multidex does not exist

### ISSUE THREAD

**Issue by Clashkid155** [flutter/flutter#98136]
## Steps to Reproduce

Have a project with over 64k modules then enable the multidex flag 
``` flutter run --multidex```
When you try running the project with the play button, you will get this error
```
/android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:8: error: package androidx.multidex does not exist
import androidx.multidex.MultiDex;
                        ^
/home/francis/AndroidStudioProjects/medel/android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:18: error: cannot find symbol
    MultiDex.install(this);
    ^
  symbol:   variable MultiDex
  location: class FlutterMultiDexApplication
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
Exception: Gradle task assembleDebug failed with exit code 1

```
But this error doesn't happen when ```flutter run``` is called from the terminal and terminal build is *2 faster than play button.

## Version info

```
[✓] Flutter (Channel stable, 2.10.0, on Manjaro Linux 5.16.5-1-MANJARO, locale en_NG.UTF-8)
    • Flutter version 2.10.0 at /home/francis/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (5 days ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francis/Android/Sdk
    • Platform android-31, build-tools 30.0.2
    • ANDROID_HOME = /opt/android-sdk
    • Java binary at: /home/francis/.local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/211.7628.21.2111.8092744/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 13.0.0
    • cmake version 3.22.2
    • ninja version 1.10.2
    • pkg-config version 1.8.0

[✓] Android Studio (version 2021.1)
    • Android Studio at /home/francis/.local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/211.7628.21.2111.8092744
    • Flutter plugin version 64.0.2
    • Dart plugin version 211.7798
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] Connected device (2 available)
    • Redmi Note 5 (mobile) • 192*****8:38796 • android-arm64 • Android 12 (API 31)
    • Linux (desktop)       • linux               • linux-x64     • Manjaro Linux 5.16.5-1-MANJARO

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

``` 

### Edit
I can only confirm this error occurs in Android Studio, although it seems it's a general issue. I just wanted to specify that.

**Comment by Clashkid155** [comment]
Okay. I can't seem to find it in flutter issues tab

**Comment by stevemessick** [comment]
https://github.com/flutter/flutter/issues/98136

**Comment by maheshmnj** [comment]
> Removing --machine from the build command below
flutter --no-color run --machine --track-widget-creation --device-id=adb-446a196d-FOatth._adb-tls-connect._tcp --multidex --dart-define=flutter.inspector.structuredErrors=true lib/main.dart

> Seems to solve the error.

Labeling this issue for further investigation.

**Comment by thaldi** [comment]
> Have same issue. After adding multidex support via terminal. (`flutter run --multidex`) Output running on from android studio:
> 
> ```
> Launching lib/main.dart on sdk gphone x86 64 arm64 in debug mode...
> Running Gradle task 'assembleDebug'...
> /home/victor/Projects/h8urs_sleep_timer/android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:8: error: package androidx.multidex does not exist
> import androidx.multidex.MultiDex;
>                         ^
> /home/victor/Projects/h8urs_sleep_timer/android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:18: error: cannot find symbol
>     MultiDex.install(this);
>     ^
>   symbol:   variable MultiDex
>   location: class FlutterMultiDexApplication
> 2 errors
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':app:compileDebugJavaWithJavac'.
> > Compilation failed; see the compiler error output for details.
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 12s
> Exception: Gradle task assembleDebug failed with exit code 1
> ```
> 
> Output running from terminal: (Everything is ok):
> 
> ```
> Building with Flutter multidex support enabled.
> Running Gradle task 'assembleDebug'...                             12.3s
> ✓  Built build/app/outputs/flutter-apk/app-debug.apk.
> W/FlutterActivityAndFragmentDelegate(10797): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps.
> Syncing files to device sdk gphone x86 64 arm64...                 755ms
> ```
> 
> 
> **Edit:** I think i solved this thanks to [StackOverflow](https://stackoverflow.com/questions/49886597/multidex-issue-with-flutter/60158352), I love it)
> 
> 1. Enable multidex.
> 
> Open `[project_folder]/app/build.gradle` and add following lines.
> 
> ```
> defaultConfig {
>     ...
> 
>     multiDexEnabled true
> }
> ```
> 
> and
> 
> ```
> dependencies {
>     ...
> 
>     implementation 'com.android.support:multidex:1.0.3'
> }
> ```
> 
> 2. Enable [Jetifier](https://developer.android.com/jetpack/androidx/migrate). _(It was enabled)_
> 
> Open `[project_folder]/android/app/gradle.properties` and add following lines.
> 
> ```
> android.useAndroidX=true
> android.enableJetifier=true
> ```

I got this error after install the google map package and this solved my issue.

**Comment by stevemessick** [comment]
@christopherfujino for visibility of flutter_tools issue with --machine and --multidex

**Comment by christopherfujino** [comment]
@GaryQian I notice that in https://github.com/flutter/flutter/commit/0407aa44804c97bcc7164818e5aabbc2db4f2fd1#diff-720367cae2c038df890f8656788545a617314a0e5af2196aef9763df6e4f3061R149 you are manually overriding `globals.terminal.usesTerminalUi` to be true. I recognize that is not the code path of this error, but why is that needed? Is this feature somehow dependent on the type of terminal? Any other ideas why this is failing with `flutter run --machine`?

**Comment by lucavenir** [comment]
I'm having this exact problem in VS Code.
I've been asked by Flutter fix to enable multidex support. Such tool added the class for me (with no manual configuration).

Now, `flutter run --multidex` works in my terminal, but it doesn't in VS Code. The following is my launch setup in VS Code
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "myapp",
      "cwd": "myapp",
      "request": "launch",
      "type": "dart",
      "args": [
        "--enable-software-rendering",
        "--multidex"
      ],
      "flutterMode": "debug",
    },
    {
      "name": "myapp (profile mode)",
      "cwd": "myapp",
      "request": "launch",
      "type": "dart",
      "args": [
        "--enable-software-rendering",
        "--multidex"
      ],
      "flutterMode": "profile"
    },
    {
      "name": "myapp (release mode)",
      "cwd": "myapp",
      "request": "launch",
      "type": "dart",
      "args": [
        "--enable-software-rendering",
        "--multidex"
      ],
      "flutterMode": "release"
    }
  ]
}
```

**Comment by MelbourneDeveloper** [comment]
I'm having this same issue, but with Android integration tests. My app runs fine, but if I try to run integration tests, I get the error

https://github.com/flutter/flutter/issues/98012#issuecomment-1270746555

**Comment by GaryQian** [comment]
@jacob314 @kenzieschmoll Routing this to you guys, It seems that IDEs specifically are not triggering the --multidex flag, thus causing builds to fail when running only in IDEs

**Comment by kenzieschmoll** [comment]
@stevemessick @helin24 @jwren

**Comment by DanTup** [comment]
There was previously a bug where `--[no-]multidex` was ignored (and not defaulted correctly) when using `--machine`. I believe I fixed that in https://github.com/flutter/flutter/pull/101689. I don't understand the error above well enough to know if these are the same issue (but based on the dates, all but the last comment - which is `flutter test` and seems to be covered by #98012 - seem like they could have been before that fix was released).

Is anyone still seeing this (with `flutter run`) on the latest stable release?

**Comment by stevemessick** [comment]
@GaryQian The OP (who was very cooperative) found that `flutter run --machine ...` caused this failure. That's why I transferred it to this repo. If users want `--multidex` they have to add it to the execution driver in the IDE, which the OP did. There's also an example in an earlier comment of a VS Code config that includes it.

It is possible Danny's patch has fixed it. I was never able to reproduce it.

**Comment by DanTup** [comment]
@Clashkid155 can you (or anyone else) confirm if you still see this on latest stable or it was fixed by #101689 and can be closed? Thanks!
